### DISCUSSION TITLE
Is there a way to find controls by class?

### DISCUSSION THREAD

**Discussion by lanyusan** [AvaloniaUI/Avalonia#9510] [Q&A]
I have been experimenting with Avalonia. 

I came from React/Web and is considering migrating an Electron desktop app to Avalonia, as it hit some performance walls, not UI but nodejs limitation in running computation intensive task.

With React, it is possible to still use DOM related api when necessary, such as `findElementById` or `findElementByClass` when sometimes it is necessary to work on visual elements directly.

I have found a few use cases if we do migrate to Avalonia that I need to find Avalonia visual controls directly by class name.

Is there a way to do so? I haven't found documentation about it.

Thanks.

**Comment by maxkatz6** [comment]
You can recursively iterate over visual tree elements with `control.GetVisualChildren()`. And you can filter control by classname, type, name or any other accessible property.

**Reply by lanyusan** [reply]
I see. Thanks
